src/Symfony/Component/Console/Application.php

```diff
@@ -117,10 +117,17 @@ public function run(InputInterface $input = null, OutputInterface $output = null
         $this->configureIO($input, $output);
 
         try {
+            $e = null;
             $exitCode = $this->doRun($input, $output);
-        } catch (\Exception $e) {
+        } catch (\Exception $x) {
+            $e = $x;
+        } catch (\Throwable $x) {
```

Comment:
The original error was rethrown before: https://github.com/symfony/symfony/pull/22435/files/6d6b04ae977d322213a4ac1bd1c84bfd32313611#diff-38eebdda7d6ebfcb7405b91167f05c9cL874

Actually, being more accurate, the original error was re-thrown **if you used an event dispatcher (and the event listeners don't set another exception)**, as I mentioned in [this issue](https://github.com/symfony/symfony/pull/20917#issuecomment-267112429).

This exception and error handling is hell and inconsistent, making it so hard to keep BC and try to handle exception and errors the same way. I still believe two different flags for exceptions & errors would have saved some headaches. 😅